wavepool/admin.py

```diff
@@ -7,6 +7,21 @@ class NewsPostForm(forms.ModelForm):
     model = NewsPost
     fields = '__all__'
 
+    def clean(self, *args, **kwargs):
+        """ cleans form data
+        """
+        # user selected "is cover story checkbox":
+        if self.data.get('is_cover_story') == 'on':
+            all_newsposts = NewsPost.objects.all()
+            # for each newspost, if it is the current cover story and NOT this newspost anyways,
+            # unset it as cover story and save
+            for n in all_newsposts:
+                if n.is_cover_story == True:
+                    if n != self.instance:
+                        n.is_cover_story = False
+            n.save()
```

Comment:
## AC1 
### wavepool/admin.py
L22 n.save()

Saving `n` outside the for loop seems to be working, if there is only one news post set as cover_story and if it is evaluated earlier in the loop.

There are some edge cases in which this function can misbehave

1. if the last news post that is evaluated is `self.instance`.
2. if two news posts are set as cover stories before this branch is merged.

In above scenarios, this function will only unset `is_cover_story` on one news post.

I recommend moving the `n.save()` inside the for loop inline with the `n.is_cover_story = False`